### DISCUSSION TITLE
Tagged PDF

### DISCUSSION THREAD

**Discussion by bigweld86** [py-pdf/pypdf#688] [Q&A]
Hi.
I need to create PDFs that are 508 compliant. My current PDF generator supports everything but Tagged PDFs, does mstamy2 support Tagged PDFs? I didn't see it in the reference documentation.
Thanks

**Comment by PatDChampionNchs** [comment]
I am also interested in the answer to this. 

I have a large document that was autotagged by Adobe Acrobat but needs over 2,000 corrections to the tagging - mostly dealing with it needing to have <TH> tags on first table rows but having <TD> tags instead. I would like to do this with PyPDF and python if possible, but when I extracted a single page and saved it through those high level library calls, the saved 1 page had no PDF tags when it should have had over a dozen.  I also don't see anything in the documentation nor, from my initial scan of the PyPDF code do I see anything obvious there either.

I have started researching PDFrw, but if that doesn't bear fruit, it is off to figuring out the Acrobat javascript API.

Tag support would be REALLY nice to have to automate 508 compliance cleanup!!

**Reply by MartinThoma** [reply]
Thank you for the feedback. I'm open to contributions, but I don't have the capacity right now to work on it myself.
